apl/aggregation-function/arg-max.mdx

```diff
@@ -0,0 +1,158 @@
+description: 'This page explains how to use the arg_max aggregation in APL.'
+---
+
+The `arg_max` aggregation in APL helps you identify the record with the maximum value for a specific numeric field and return one or more additional fields from that record. Use `arg_max` when you want to determine key details associated with a record having the maximum value, such as the longest request duration, highest transaction amount, or most significant span duration.
+
+This aggregation is particularly useful in scenarios like:
+
+- Pinpointing the slowest HTTP requests in log data.
+- Identifying the longest span durations in OpenTelemetry traces.
+- Highlighting the highest severity security alerts in logs.
+
+## For users of other query languages
+
+If you come from other query languages, this section explains how to adjust your existing queries to achieve the same results in APL.
+
+<AccordionGroup>
+<Accordion title="Splunk SPL users">
+
+In Splunk SPL, you use `stats` with a combination of `max` and `by` clauses to achieve similar results. APL provides a dedicated `arg_max` aggregation that simplifies this process.
+
+<CodeGroup>
+```sql Splunk example
+| stats max(req_duration_ms) as max_duration by id, uri
+```
+
+```kusto APL equivalent
+['sample-http-logs']
+| summarize arg_max(req_duration_ms, id, uri)
+```
+</CodeGroup>
+
+</Accordion>
+<Accordion title="ANSI SQL users">
+
+In ANSI SQL, you typically use a subquery to find the maximum value and then join it back to the original table to retrieve additional fields. APL’s `arg_max` provides a more concise and efficient alternative.
+
+<CodeGroup>
+```sql SQL example
+WITH MaxValues AS (
+    SELECT id, MAX(req_duration_ms) as max_duration
+    FROM sample_http_logs
+    GROUP BY id
+)
+SELECT logs.id, logs.uri, MaxValues.max_duration
+FROM sample_http_logs logs
+JOIN MaxValues
+ON logs.id = MaxValues.id;
+```
+
+```kusto APL equivalent
+['sample-http-logs']
+| summarize arg_max(req_duration_ms, id, uri)
+```
+</CodeGroup>
+
+</Accordion>
+</AccordionGroup>
+
+## Usage
+
+### Syntax
+
+```kusto
+| summarize arg_max(numeric_field, field1[, field2, ...])
+```
+
+### Parameters
+
+| Parameter        | Description                                                                         |
+|------------------|-------------------------------------------------------------------------------------|
+| `numeric_field`  | The numeric field whose maximum value determines the selected record.             |
+| `field1, field2` | The additional fields to retrieve from the record with the maximum numeric value. |
+
+### Returns
+
+`arg_max` returns a row for each group (or the entire dataset if no grouping is specified), containing the fields specified in the query.
+
+## Use case examples
+
+<Tabs>
+<Tab title="Log analysis">
+
+Find the slowest HTTP request for each URI in the `['sample-http-logs']` dataset.
+
+**Query**
+
+```kusto
+['sample-http-logs']
+| summarize arg_max(req_duration_ms, method) by uri
```

Comment:
IMO a better example would be with swapped uri & method - ie arg_max(duration, uri) by method - which would return slowest paths for any given HTTP method.

> Find the slowest HTTP request for each URI in the `['sample-http-logs']` dataset.
that doesn't need arg_max, would be just `summarize max(duration) by uri`

Reply:
or perhaps replace the method with request_id, that also makes sense